decoder/reference_origins.go

```diff
@@ -111,6 +111,14 @@ func (d *PathDecoder) referenceOriginsInBody(body hcl.Body, bodySchema *schema.B
 			}
 		}
 
+		if aSchema.IsDepKey && bodySchema.LinksTo != nil {
```

Comment:
It _feels_ like this is not guarded enough and might introduce `DirectOrigin`s at unwanted locations.

Reply:
Furthermore, all origins would link to the same location, and we don't support having multiple links inside a body.